mongodbatlas/framework/conversion/string.go

```diff
@@ -17,3 +17,12 @@ func TypesListToString(ctx context.Context, set types.List) []string {
 	_ = set.ElementsAs(ctx, &results, false)
 	return results
 }
+
+// StringValueToFramework converts a string value to a Framework String value.
+// An empty string is converted to a null String. Useful for optional attributes.
+func StringValueToFramework(v string) types.String {
+	if v == "" {
+		return types.StringNull()
```

Comment:
why we cannot use types.StringValue(v) when v is `""`?

Reply:
I believe that's because in SDKv2 there was no concept of null values for string, ints etc so they always stored "" (for strings) or 0 (for integers) whether or not the user provided value for that attribute in the config. 
But now the Framework expects the state to be same as config and if the user has not specified an attribute in the config it is expected to be null in the state (sometimes we get "" for some values from the API, so Terraform throws inconsistent values error)
@AgustinBettati correct me if I'm wrong.

Reply:
Correct, this is relevant primarily for migration tests. If the empty string is defined directly you end up with non-empty plan diffs (related issue: https://github.com/hashicorp/terraform/issues/31887).

Reply:
Thanks for the clarifications here